### ISSUE TITLE
OS X Toolchain no longer works on High Sierra

### ISSUE THREAD

**Issue by jbf** [nerves-project/toolchains#57]
After upgrading to latest nerves I can no longer compile on my OS X host.

### Environment

$ elixir -v
Erlang/OTP 22 [erts-10.7] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

Elixir 1.10.2 (compiled with Erlang/OTP 22)

$ mix nerves.env --info
|nerves_bootstrap| Environment Package List

  No packages found
|nerves_bootstrap| Loadpaths Start

Nerves environment
  MIX_TARGET:   host
  MIX_ENV:      dev

NERVES_SYSTEM is unset
NERVES_TOOLCHAIN is unset
|nerves_bootstrap| Environment Variable List
  target:     host
  toolchain:  unset
  system:     unset
  app:        /Users/jbf/src/minimal_nerves

|nerves_bootstrap| Loadpaths End

* Additional information about your host, target hardware or environment that
  may help

$ uname -a
Darwin Joels-iMac 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64

High Sierra 10.13.6

### Current behavior

$ MIX_TARGET=rpi0 mix firmware

Nerves environment
  MIX_TARGET:   rpi0
  MIX_ENV:      dev

==> nerves_runtime
mkdir -p /Users/jbf/src/minimal_nerves/_build/rpi0_dev/lib/nerves_runtime/obj
mkdir -p /Users/jbf/src/minimal_nerves/_build/rpi0_dev/lib/nerves_runtime/priv
/Users/jbf/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.3.1/bin/armv6-rpi-linux-gnueabi-gcc -c -I/Users/jbf/.nerves/artifacts/nerves_system_rpi0-portable-1.11.0/staging/usr/lib/erlang/erts-10.6.4/include -I/Users/jbf/.nerves/artifacts/nerves_system_rpi0-portable-1.11.0/staging/usr/lib/erlang/lib/erl_interface-3.13.1/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/jbf/.nerves/artifacts/nerves_system_rpi0-portable-1.11.0/staging/usr/include -std=gnu99 -o /Users/jbf/src/minimal_nerves/_build/rpi0_dev/lib/nerves_runtime/obj/nerves_runtime.o src/nerves_runtime.c
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/jbf/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.3.1/bin/armv6-rpi-linux-gnueabi-gcc (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/jbf/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.3.1/bin/armv6-rpi-linux-gnueabi-gcc (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

make: *** [/Users/jbf/src/minimal_nerves/_build/rpi0_dev/lib/nerves_runtime/obj/nerves_runtime.o] Abort trap: 6
could not compile dependency :nerves_runtime, "mix compile" failed. You can recompile this dependency with "mix deps.compile nerves_runtime", update it with "mix deps.update nerves_runtime" or clean it with "mix deps.clean nerves_runtime"
==> minimal_nerves
** (Mix) Could not compile with "make" (exit status: 2).
You need to have gcc and make installed. Try running the
commands "gcc --version" and / or "make --version". If these programs
are not installed, you will be prompted to install them.

### Expected behavior

A compiled project

**Comment by mcrumm** [comment]
Samesies 😭 

```
$ MIX_TARGET=rpi mix firmware

Nerves environment
  MIX_TARGET:   rpi
  MIX_ENV:      dev

==> nerves_runtime
mkdir -p /Users/mike/Code/mcrumm/mr_color_switch/_build/rpi_dev/lib/nerves_runtime/obj
mkdir -p /Users/mike/Code/mcrumm/mr_color_switch/_build/rpi_dev/lib/nerves_runtime/priv
/Users/mike/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.3.1/bin/armv6-rpi-linux-gnueabi-gcc -c -I/Users/mike/.nerves/artifacts/nerves_system_rpi-portable-1.11.0/staging/usr/lib/erlang/erts-10.6.4/include -I/Users/mike/.nerves/artifacts/nerves_system_rpi-portable-1.11.0/staging/usr/lib/erlang/lib/erl_interface-3.13.1/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/mike/.nerves/artifacts/nerves_system_rpi-portable-1.11.0/staging/usr/include -std=gnu99 -o /Users/mike/Code/mcrumm/mr_color_switch/_build/rpi_dev/lib/nerves_runtime/obj/nerves_runtime.o src/nerves_runtime.c
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/mike/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.3.1/bin/armv6-rpi-linux-gnueabi-gcc (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/mike/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.3.1/bin/armv6-rpi-linux-gnueabi-gcc (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

make: *** [/Users/mike/Code/mcrumm/mr_color_switch/_build/rpi_dev/lib/nerves_runtime/obj/nerves_runtime.o] Abort trap: 6
```

**Comment by mcrumm** [comment]
Hi @fhunleth @mobileoverlord ! 

I've been away from Nerves for a little bit, so apologies if this is already obvious, but if it helps for debugging, I had to downgrade to `nerves_bootstrap 1.7.1` and then explicitly pin the targets to `~> 1.10.0` to get a successful build on High Sierra.

If there's anything I can do to help in the debugging, please let me know. Thanks for all of the amazing work on the Nerves platform! ❤️

**Comment by mobileoverlord** [comment]
Thanks for posting the current work around. The issue is that Apple made accessing of previous OS install images practically impossible. The toolchains need to be built on an older version of macOS and accessing one of these machines is difficult. I am working on trying to get the toolchain to build under the macOS executor on Circle.

**Comment by mobileoverlord** [comment]
I finally got a High Sierra virtual machine running locally. I am going to try to build the release toolchains and re upload them. I'll let you know when its ready.

**Comment by fhunleth** [comment]
Closing - fixed in 1.3.2 release.
